selfdrive/controls/controlsd.py

```diff
@@ -650,6 +651,12 @@ def state_control(self, CS):
         cloudlog.error(f"actuators.{p} not finite {actuators.to_dict()}")
         setattr(actuators, p, 0.0)
 
+    # decrement personality on distance button press
+    if self.CP.openpilotLongitudinalControl:
+      if any(not be.pressed and be.type == ButtonType.gapAdjustCruise for be in CS.buttonEvents):
+        self.personality = (self.personality - 1) % 3
+        self.params.put('LongitudinalPersonality', str(self.personality))
```

Comment:
this has to be nonblocking. it should also be published. params shouldn't really be used as IPC

Reply:
It is published and we can use it in the UI, but then nothing will save the change

Reply:
Made UI listen to controlsState.personality and update when it changes, but only if it's visible. When it's not, it relies on the show event and the parameter to update. How does this look?

Reply:
If you're fine with the toggles panel not updating live, we can remove all this and only have it refresh on widget open for now!